

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>5. NGS - Genome assembly &mdash; Documentation</title>
  

  
  
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../_static/css/seb.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="Documentation" href="../index.html"/>
        <link rel="next" title="6. NGS - Genome annotation" href="../ngs-annotation/index.html"/>
        <link rel="prev" title="4. NGS - Taxonomic investigation" href="../ngs-taxonomic-investigation/index.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> Genomics Tutorial
          

          
            
            <img src="../_static/icon.png" class="logo" />
          
          </a>

          
            
            
              <div class="version">
                2017.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../cli/index.html">1. Command-line interface introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-tools/index.html">2. NGS - Tool installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-qc/index.html">3. NGS - Quality control</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-taxonomic-investigation/index.html">4. NGS - Taxonomic investigation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">5. NGS - Genome assembly</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preface">5.1. Preface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#learning-outcomes">5.2. Learning outcomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#before-we-start">5.3. Before we start</a></li>
<li class="toctree-l2"><a class="reference internal" href="#creating-a-genome-assembly">5.4. Creating a genome assembly</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#installing-the-software">5.4.1. Installing the software</a></li>
<li class="toctree-l3"><a class="reference internal" href="#spades-usage">5.4.2. SPAdes usage</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#assembly-quality-assessment">5.5. Assembly quality assessment</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#assembly-statistics">5.5.1. Assembly statistics</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#assemblathon">5.6. Assemblathon</a></li>
<li class="toctree-l2"><a class="reference internal" href="#further-reading">5.7. Further reading</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#background-on-genome-assemblies">5.7.1. Background on Genome Assemblies</a></li>
<li class="toctree-l3"><a class="reference internal" href="#evaluation-of-genome-assembly-software">5.7.2. Evaluation of Genome Assembly Software</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#web-links">5.8. Web links</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-annotation/index.html">6. NGS - Genome annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-mapping/index.html">7. NGS - Read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-variantcalling/index.html">8. NGS - Variant calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="../general/quickref.html">9. Quick command reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../general/downloads.html">10. Downloads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../general/references.html">11. References</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">Genomics Tutorial</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>5. NGS - Genome assembly</li>
      <li class="wy-breadcrumbs-aside">
        
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="ngs-genome-assembly">
<span id="ngs-assembly"></span><h1>5. NGS - Genome assembly<a class="headerlink" href="#ngs-genome-assembly" title="Permalink to this headline">¶</a></h1>
<div class="section" id="preface">
<h2>5.1. Preface<a class="headerlink" href="#preface" title="Permalink to this headline">¶</a></h2>
<p>In this section we will use our skill on the command-line interface to create a
genome assembly from sequencing data.</p>
<p>There is an accompanying lecture for this tutorial:</p>
<ul class="simple">
<li><a class="reference external" href="https://dx.doi.org/10.6084/m9.figshare.2972323.v1">Genome Assembly: An Introduction</a> available at
<a class="reference external" href="https://dx.doi.org/10.6084/m9.figshare.2972323.v1">figshare</a>
(<a class="reference external" href="https://dx.doi.org/10.6084/m9.figshare.2972323.v1">https://dx.doi.org/10.6084/m9.figshare.2972323.v1</a>).</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You will encounter some <strong>To-do</strong> sections at times. Write the solutions and answers into a text-file.</p>
</div>
</div>
<div class="section" id="learning-outcomes">
<h2>5.2. Learning outcomes<a class="headerlink" href="#learning-outcomes" title="Permalink to this headline">¶</a></h2>
<p>After studying this tutorial you should be able to:</p>
<ol class="arabic simple">
<li>Compute and interpret a whole genome assembly.</li>
<li>Judge the quality of a genome assembly.</li>
</ol>
</div>
<div class="section" id="before-we-start">
<h2>5.3. Before we start<a class="headerlink" href="#before-we-start" title="Permalink to this headline">¶</a></h2>
<p>Lets see how our directory structure looks so far:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>
<span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">F</span>
</pre></div>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">data</span><span class="o">/</span>
<span class="n">kraken</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">++</span>
<span class="n">trimmed</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">fastqc</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">solexaqa</span><span class="o">/</span>
</pre></div>
</div>
</div>
<div class="section" id="creating-a-genome-assembly">
<h2>5.4. Creating a genome assembly<a class="headerlink" href="#creating-a-genome-assembly" title="Permalink to this headline">¶</a></h2>
<p>We want to create a genome assembly for our ancestor.
We are going to use the quality trimmed forward and backward DNA sequences and
use a program called <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> to build a genome assembly.</p>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Discuss briefly why we are using the ancestral sequences to create a
reference genome as opposed to the evolved line.</li>
</ol>
</div>
<div class="section" id="installing-the-software">
<h3>5.4.1. Installing the software<a class="headerlink" href="#installing-the-software" title="Permalink to this headline">¶</a></h3>
<p>We are going to use a program called <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> fo assembling our genome.
In a recent evaluation of assembly software, <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> was found to be a good
choice for fungal genomes <a class="reference internal" href="../general/references.html#abbas2014" id="id1">[ABBAS2014]</a>.
It is also simple to install and use.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">ngs</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">spades</span>
</pre></div>
</div>
</div>
<div class="section" id="spades-usage">
<h3>5.4.2. <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> usage<a class="headerlink" href="#spades-usage" title="Permalink to this headline">¶</a></h3>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># change to your analysis root folder</span>
<span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>

<span class="c1"># first create a output directory for the assemblies</span>
<span class="n">mkdir</span> <span class="n">assembly</span>

<span class="c1"># to get a help for spades and an overview of the parameter type:</span>
<span class="n">spades</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">h</span>
</pre></div>
</div>
<p>The two files we need to submit to <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> are two paired-end read files.</p>
<p class="sebcode">spades.py -o assembly/spades-default/ -1 trimmed/ancestor-R1.fastq.trimmed.gz -2 trimmed/ancestor-R2.fastq.trimmed.gz</p>
<div class="admonition-todo admonition" id="index-1">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Run <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> with default parameters on the ancestor</li>
<li>Read in the <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> manual about about assembling with 2x150bp reads</li>
<li>Run <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> a second time but use the options suggested at the <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> manual <a class="reference external" href="http://spades.bioinf.spbau.ru/release3.9.1/manual.html#sec3.4">section 3.4</a> for assembling 2x150bp paired-end reads (are fungi multicellular?). Use a different output directory <code class="docutils literal"><span class="pre">assembly/spades-150</span></code> for this run.</li>
</ol>
</div>
<div class="admonition hint">
<p class="first admonition-title">Hint</p>
<p class="last">Should you not get it right, try these commands <a class="reference external" href="../_static/code/assembly1.txt">here</a>.</p>
</div>
</div>
</div>
<div class="section" id="assembly-quality-assessment">
<h2>5.5. Assembly quality assessment<a class="headerlink" href="#assembly-quality-assessment" title="Permalink to this headline">¶</a></h2>
<div class="section" id="assembly-statistics">
<h3>5.5.1. Assembly statistics<a class="headerlink" href="#assembly-statistics" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> (QUality ASsesment Tool) <a class="reference internal" href="../general/references.html#gurevich2013" id="id2">[GUREVICH2013]</a>, evaluates genome assemblies by computing various metrics, including:</p>
<ul class="simple">
<li>N50: length for which the collection of all contigs of that length or
longer covers at least 50% of assembly length</li>
<li>NG50: where length of the reference genome is being covered</li>
<li>NA50 and NGA50: where aligned blocks instead of contigs are taken</li>
<li>missassemblies: misassembled and unaligned contigs or contigs bases</li>
<li>genes and operons covered</li>
</ul>
<p>It is easy with <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> to compare these measures among several assemblies.
The program can be used on their website (<a class="reference external" href="http://quast.bioinf.spbau.ru/">http://quast.bioinf.spbau.ru/</a>).</p>
<p>We can install it locally with:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">ngs</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">quast</span>
</pre></div>
</div>
<p>Run <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> with both assembly scaffolds.fasta files to compare the results.</p>
<div class="admonition hint">
<p class="first admonition-title">Hint</p>
<p class="last">Should you be unable to run <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> on the data, you can download the assemblies <a class="reference external" href="http://compbio.massey.ac.nz/data/203341/assembly.tar.gz">here</a>. Unarchive and uncompress the files with <code class="docutils literal"><span class="pre">tar</span> <span class="pre">-xvzf</span> <span class="pre">assembly.tar.gz</span></code>.</p>
</div>
<p class="sebcode">quast -o assembly/quast assembly/spades-default/scaffolds.fasta assembly/spades-150/scaffolds.fasta</p>
<div class="admonition-todo admonition" id="index-2">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Compare the results of <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> with regards to the two different assemblies.</li>
<li>Which one do you prefer and why?</li>
</ol>
</div>
</div>
</div>
<div class="section" id="assemblathon">
<h2>5.6. Assemblathon<a class="headerlink" href="#assemblathon" title="Permalink to this headline">¶</a></h2>
<div class="admonition-todo admonition" id="index-3">
<p class="first admonition-title">Todo</p>
<p class="last">Now that you know the basics for assembling a genome and judging their quality, play with the <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> parameters to create the best assembly possible.</p>
</div>
<div class="admonition-todo admonition" id="index-4">
<p class="first admonition-title">Todo</p>
<p class="last">Once you have your final assembly, rename your assembly directory int <code class="docutils literal"><span class="pre">spades-final</span></code>, e.g. <code class="docutils literal"><span class="pre">mv</span> <span class="pre">assembly/spades-default</span> <span class="pre">assembly/spades-final</span></code>.</p>
</div>
</div>
<div class="section" id="further-reading">
<h2>5.7. Further reading<a class="headerlink" href="#further-reading" title="Permalink to this headline">¶</a></h2>
<div class="section" id="background-on-genome-assemblies">
<h3>5.7.1. Background on Genome Assemblies<a class="headerlink" href="#background-on-genome-assemblies" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>How to apply de Bruijn graphs to genome assembly. <a class="reference internal" href="../general/references.html#compeau2011" id="id3">[COMPEAU2011]</a></li>
<li>Sequence assembly demystified. <a class="reference internal" href="../general/references.html#nagarajan2013" id="id4">[NAGARAJAN2013]</a></li>
</ul>
</div>
<div class="section" id="evaluation-of-genome-assembly-software">
<h3>5.7.2. Evaluation of Genome Assembly Software<a class="headerlink" href="#evaluation-of-genome-assembly-software" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>GAGE: A critical evaluation of genome assemblies and assembly algorithms. <a class="reference internal" href="../general/references.html#salzberg2012" id="id5">[SALZBERG2012]</a></li>
<li>Assessment of de novo assemblers for draft genomes: a case study with fungal genomes. <a class="reference internal" href="../general/references.html#abbas2014" id="id6">[ABBAS2014]</a></li>
</ul>
</div>
</div>
<div class="section" id="web-links">
<h2>5.8. Web links<a class="headerlink" href="#web-links" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Lectures for this topic: <a class="reference external" href="https://dx.doi.org/10.6084/m9.figshare.2972323.v1">Genome Assembly: An Introduction</a></li>
<li><a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a></li>
<li><a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a></li>
<li><a class="reference external" href="https://rrwick.github.io/Bandage/">Bandage</a> (Bioinformatics Application for Navigating De novo Assembly Graphs Easily) is a program that visualizes a genome assembly as a graph <a class="reference internal" href="../general/references.html#wick2015" id="id7">[WICK2015]</a>.</li>
</ul>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../ngs-annotation/index.html" class="btn btn-neutral float-right" title="6. NGS - Genome annotation" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ngs-taxonomic-investigation/index.html" class="btn btn-neutral" title="4. NGS - Taxonomic investigation" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2017, Sebastian Schmeier.
      Last updated on Jan 29, 2017.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'2017.1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>