

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>7. Variant calling</title>
  

  
  
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../_static/css/seb.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="" href="../index.html"/>
        <link rel="next" title="8. Genome annotation" href="../ngs-annotation/index.html"/>
        <link rel="prev" title="6. Taxonomic investigation" href="../ngs-taxonomic-investigation/index.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> Genomics Tutorial
          

          
            
            <img src="../_static/icon.png" class="logo" />
          
          </a>

          
            
            
              <div class="version">
                2017.10
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">1. Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-tools/index.html">2. Tool installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-qc/index.html">3. Quality control</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-assembly/index.html">4. Genome assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-mapping/index.html">5. Read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-taxonomic-investigation/index.html">6. Taxonomic investigation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">7. Variant calling</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preface">7.1. Preface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#overview">7.2. Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#learning-outcomes">7.3. Learning outcomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#before-we-start">7.4. Before we start</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing-necessary-software">7.5. Installing necessary software</a></li>
<li class="toctree-l2"><a class="reference internal" href="#preprocessing">7.6. Preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#calling-variants">7.7. Calling variants</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#samtools-mpileup">7.7.1. SAMtools mpileup</a></li>
<li class="toctree-l3"><a class="reference internal" href="#freebayes">7.7.2. Freebayes</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#post-processing">7.8. Post-processing</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#understanding-the-output-files-vcf">7.8.1. Understanding the output files (.vcf)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#statistics">7.8.2. Statistics</a></li>
<li class="toctree-l3"><a class="reference internal" href="#variant-filtration">7.8.3. Variant filtration</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-annotation/index.html">8. Genome annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-orthology/index.html">9. Orthology and Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-voi/index.html">10. Variants-of-interest</a></li>
<li class="toctree-l1"><a class="reference internal" href="../quickref.html">11. Quick command reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../code.html">12. Coding solutions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../downloads.html">13. Downloads</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Genomics Tutorial</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>7. Variant calling</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="variant-calling">
<span id="ngs-variantcalling"></span><h1>7. Variant calling<a class="headerlink" href="#variant-calling" title="Permalink to this headline">¶</a></h1>
<div class="section" id="preface">
<h2>7.1. Preface<a class="headerlink" href="#preface" title="Permalink to this headline">¶</a></h2>
<p>In this section we will use our genome assembly based on the ancestor and call genetic variants in the evolved line <a class="reference internal" href="#nielsen2011" id="id1">[NIELSEN2011]</a>.</p>
</div>
<div class="section" id="overview">
<h2>7.2. Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>The part of the workflow we will work on in this section can be viewed in <a class="reference internal" href="#fig-workflow-var"><span class="std std-numref">Fig. 7.1</span></a>.</p>
<div class="figure" id="id3">
<span id="fig-workflow-var"></span><img alt="../_images/workflow6.png" src="../_images/workflow6.png" />
<p class="caption"><span class="caption-number">Fig. 7.1 </span><span class="caption-text">The part of the workflow we will work on in this section marked in red.</span></p>
</div>
</div>
<div class="section" id="learning-outcomes">
<h2>7.3. Learning outcomes<a class="headerlink" href="#learning-outcomes" title="Permalink to this headline">¶</a></h2>
<p>After studying this tutorial section you should be able to:</p>
<p>#. Use tools to call variants based on a reference genome.
#, Be able to describe what influences the calling of variants.</p>
</div>
<div class="section" id="before-we-start">
<h2>7.4. Before we start<a class="headerlink" href="#before-we-start" title="Permalink to this headline">¶</a></h2>
<p>Lets see how our directory structure looks so far:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>
<span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">F</span>
</pre></div>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">assembly</span><span class="o">/</span>
<span class="n">data</span><span class="o">/</span>
<span class="n">kraken</span><span class="o">/</span>
<span class="n">mappings</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">++</span>
<span class="n">trimmed</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">fastqc</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">solexaqa</span><span class="o">/</span>
</pre></div>
</div>
</div>
<div class="section" id="installing-necessary-software">
<h2>7.5. Installing necessary software<a class="headerlink" href="#installing-necessary-software" title="Permalink to this headline">¶</a></h2>
<p>Tools we are going to use in this section and how to intall them if you not have done it yet.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># activate the env</span>
<span class="n">source</span> <span class="n">activate</span> <span class="n">ngs</span>

<span class="c1"># Install these tools into the conda environment</span>
<span class="c1"># if not already installed</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">samtools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">bamtools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">freebayes</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">bedtools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">vcflib</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">rtg</span><span class="o">-</span><span class="n">tools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">bcftools</span>
</pre></div>
</div>
</div>
<div class="section" id="preprocessing">
<h2>7.6. Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">¶</a></h2>
<p>We first need to make an index of our reference genome as this is required by the SNP caller.
Given a scaffold/contig file in fasta-format, e.g. <code class="docutils literal"><span class="pre">scaffolds.fasta</span></code> which is located in the directory <code class="docutils literal"><span class="pre">assembly/spades_final</span></code>, use <a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a> to do this:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">faidx</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="n">final</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Furthermore we need to pre-process our mapping files a bit further and create a bam-index file (<code class="docutils literal"><span class="pre">.bai</span></code>) for the bam-file we want to work with:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">bamtools</span> <span class="n">index</span> <span class="o">-</span><span class="ow">in</span> <span class="n">mappings</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">sorted</span><span class="o">.</span><span class="n">concordant</span><span class="o">.</span><span class="n">q20</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Lets also create a new directory for the variants:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">variants</span>
</pre></div>
</div>
</div>
<div class="section" id="calling-variants">
<h2>7.7. Calling variants<a class="headerlink" href="#calling-variants" title="Permalink to this headline">¶</a></h2>
<div class="section" id="samtools-mpileup">
<h3>7.7.1. SAMtools mpileup<a class="headerlink" href="#samtools-mpileup" title="Permalink to this headline">¶</a></h3>
<p>We use the sorted filtered bam-file that we produced in the mapping step before.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># We first pile up all the reads and then call variants</span>
<span class="n">samtools</span> <span class="n">mpileup</span> <span class="o">-</span><span class="n">u</span> <span class="o">-</span><span class="n">g</span> <span class="o">-</span><span class="n">f</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="n">final</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span> <span class="n">mappings</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">sorted</span><span class="o">.</span><span class="n">concordant</span><span class="o">.</span><span class="n">q20</span><span class="o">.</span><span class="n">bam</span> <span class="o">|</span> <span class="n">bcftools</span> <span class="n">call</span> <span class="o">-</span><span class="n">v</span> <span class="o">-</span><span class="n">m</span> <span class="o">-</span><span class="n">O</span> <span class="n">z</span> <span class="o">-</span><span class="n">o</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p><a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a> mpileup parameter:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-u</span></code>: uncompressed output</li>
<li><code class="docutils literal"><span class="pre">-g</span></code>: generate genotype likelihoods in BCF format</li>
<li><code class="docutils literal"><span class="pre">-f</span> <span class="pre">FILE</span></code>: faidx indexed reference sequence file</li>
</ul>
<p><a class="reference external" href="http://www.htslib.org/doc/bcftools.html">BCFtools</a> view parameter:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-v</span></code>: output variant sites only</li>
<li><code class="docutils literal"><span class="pre">-m</span></code>: alternative model for multiallelic and rare-variant calling</li>
<li><code class="docutils literal"><span class="pre">-o</span></code>: output file-name</li>
<li><code class="docutils literal"><span class="pre">-O</span> <span class="pre">z</span></code>: output type: ‘z’ compressed VCF</li>
</ul>
</div>
<div class="section" id="freebayes">
<h3>7.7.2. Freebayes<a class="headerlink" href="#freebayes" title="Permalink to this headline">¶</a></h3>
<p>As an alternative we can do some variant calling with another tool called <a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a>.
Given a reference genome scaffold file in fasta-format, e.g. <code class="docutils literal"><span class="pre">scaffolds.fasta</span></code> and the index in <code class="docutils literal"><span class="pre">.fai</span></code> format and a mapping file (.bam file) and a mapping index (.bai file), we can call variants with <a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a> like so:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># Now we call variants and pipe the results into a new file</span>
<span class="n">freebayes</span> <span class="o">-</span><span class="n">f</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="n">final</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span> <span class="n">mappings</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">sorted</span><span class="o">.</span><span class="n">concordant</span><span class="o">.</span><span class="n">q20</span><span class="o">.</span><span class="n">bam</span> <span class="o">|</span> <span class="n">gzip</span> <span class="o">&gt;</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">freebayes</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="post-processing">
<h2>7.8. Post-processing<a class="headerlink" href="#post-processing" title="Permalink to this headline">¶</a></h2>
<div class="section" id="understanding-the-output-files-vcf">
<h3>7.8.1. Understanding the output files (.vcf)<a class="headerlink" href="#understanding-the-output-files-vcf" title="Permalink to this headline">¶</a></h3>
<p>Lets look at a vcf-file:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># first 10 lines, which are part of the header</span>
<span class="n">zcat</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">|</span> <span class="n">head</span>
</pre></div>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1">##fileformat=VCFv4.2</span>
<span class="c1">##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;</span>
<span class="c1">##samtoolsVersion=1.3.1+htslib-1.3.1</span>
<span class="c1">##samtoolsCommand=samtools mpileup -g -f assembly/spades-final/scaffolds.fasta -o variants/evolved-6.mpileup.bcf mappings/evolved-6.sorted.concordant.q20.bam</span>
<span class="c1">##reference=file://assembly/spades-final/scaffolds.fasta</span>
<span class="c1">##contig=&lt;ID=NODE_1_length_1419525_cov_15.3898,length=1419525&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_2_length_1254443_cov_15.4779,length=1254443&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_3_length_972329_cov_15.3966,length=972329&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_4_length_951685_cov_15.4231,length=951685&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_5_length_925222_cov_15.39,length=925222&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_6_length_916533_cov_15.4426,length=916533&gt;</span>
</pre></div>
</div>
<p>Lets look at the variants:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># remove header lines and look at top 4 entires</span>
<span class="n">zcat</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">|</span> <span class="n">egrep</span> <span class="o">-</span><span class="n">v</span> <span class="s1">&#39;##&#39;</span> <span class="o">|</span> <span class="n">head</span> <span class="o">-</span><span class="mi">4</span>
</pre></div>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1">#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  mappings/evolved-6.sorted.concordant.q20.bam</span>
<span class="n">NODE_1_length_1419525_cov_15</span><span class="o">.</span><span class="mi">3898</span>       <span class="mi">24721</span>   <span class="o">.</span>       <span class="n">T</span>       <span class="n">C</span>       <span class="mi">164</span>     <span class="o">.</span>       <span class="n">DP</span><span class="o">=</span><span class="mi">12</span><span class="p">;</span><span class="n">VDB</span><span class="o">=</span><span class="mf">0.205941</span><span class="p">;</span><span class="n">SGB</span><span class="o">=-</span><span class="mf">0.680642</span><span class="p">;</span><span class="n">MQ0F</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">AC</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP4</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mi">40</span>     <span class="n">GT</span><span class="p">:</span><span class="n">PL</span>   <span class="mi">1</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">191</span><span class="p">,</span><span class="mi">36</span><span class="p">,</span><span class="mi">0</span>
<span class="n">NODE_1_length_1419525_cov_15</span><span class="o">.</span><span class="mi">3898</span>       <span class="mi">157033</span>  <span class="o">.</span>       <span class="n">AAGAGAGAGAGAGAGAGAGAGAGA</span>        <span class="n">AAGAGAGAGAGAGAGAGAGAGA</span>  <span class="mf">39.3328</span>  <span class="o">.</span>       <span class="n">INDEL</span><span class="p">;</span><span class="n">IDV</span><span class="o">=</span><span class="mi">6</span><span class="p">;</span><span class="n">IMF</span><span class="o">=</span><span class="mf">0.146341</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">41</span><span class="p">;</span><span class="n">VDB</span><span class="o">=</span><span class="mf">0.0813946</span><span class="p">;</span><span class="n">SGB</span><span class="o">=-</span><span class="mf">0.616816</span><span class="p">;</span><span class="n">MQSB</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MQ0F</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ICB</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">HOB</span><span class="o">=</span><span class="mf">0.5</span><span class="p">;</span><span class="n">AC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP4</span><span class="o">=</span><span class="mi">13</span><span class="p">,</span><span class="mi">17</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">3</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mi">42</span>     <span class="n">GT</span><span class="p">:</span><span class="n">PL</span>   <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">75</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">255</span>
<span class="n">NODE_1_length_1419525_cov_15</span><span class="o">.</span><span class="mi">3898</span>       <span class="mi">162469</span>  <span class="o">.</span>       <span class="n">T</span>       <span class="n">C</span>       <span class="mf">19.609</span>  <span class="o">.</span>       <span class="n">DP</span><span class="o">=</span><span class="mi">16</span><span class="p">;</span><span class="n">VDB</span><span class="o">=</span><span class="mf">0.045681</span><span class="p">;</span><span class="n">SGB</span><span class="o">=-</span><span class="mf">0.511536</span><span class="p">;</span><span class="n">RPB</span><span class="o">=</span><span class="mf">0.032027</span><span class="p">;</span><span class="n">MQB</span><span class="o">=</span><span class="mf">0.832553</span><span class="p">;</span><span class="n">BQB</span><span class="o">=</span><span class="mf">0.130524</span><span class="p">;</span><span class="n">MQ0F</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ICB</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">HOB</span><span class="o">=</span><span class="mf">0.5</span><span class="p">;</span><span class="n">AC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">DP4</span><span class="o">=</span><span class="mi">13</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mi">39</span>        <span class="n">GT</span><span class="p">:</span><span class="n">PL</span>   <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">54</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">155</span>
</pre></div>
</div>
<p>The fields in a vcf-file are described in he table (<a class="reference internal" href="#table-vcf"><span class="std std-numref">Table 7.1</span></a>) below:</p>
<span id="table-vcf"></span><table border="1" class="docutils" id="id4">
<caption><span class="caption-number">Table 7.1 </span><span class="caption-text">The vcf-file format fields.</span><a class="headerlink" href="#id4" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="5%" />
<col width="11%" />
<col width="84%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Col</th>
<th class="head">Field</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>1</td>
<td>CHROM</td>
<td>Chromosome name</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>POS</td>
<td>1-based position. For an indel, this is the position preceding the indel.</td>
</tr>
<tr class="row-even"><td>3</td>
<td>ID</td>
<td>Variant identifier. Usually the dbSNP rsID.</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>REF</td>
<td>Reference sequence at POS involved in the variant. For a SNP, it is a single base.</td>
</tr>
<tr class="row-even"><td>5</td>
<td>ALT</td>
<td>Comma delimited list of alternative seuqence(s).</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>QUAL</td>
<td>Phred-scaled probability of all samples being homozygous reference.</td>
</tr>
<tr class="row-even"><td>7</td>
<td>FILTER</td>
<td>Semicolon delimited list of filters that the variant fails to pass.</td>
</tr>
<tr class="row-odd"><td>8</td>
<td>INFO</td>
<td>Semicolon delimited list of variant information.</td>
</tr>
<tr class="row-even"><td>9</td>
<td>FORMAT</td>
<td>Colon delimited list of the format of individual genotypes in the following fields.</td>
</tr>
<tr class="row-odd"><td>10+</td>
<td>Sample(s)</td>
<td>Individual genotype information defined by FORMAT.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="statistics">
<h3>7.8.2. Statistics<a class="headerlink" href="#statistics" title="Permalink to this headline">¶</a></h3>
<p>Now we can use it to do some statistics and filter our variant calls.</p>
<p>First, to prepare out vcf-file for querying we need to index it with <code class="docutils literal"><span class="pre">tabix</span></code>:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">tabix</span> <span class="o">-</span><span class="n">p</span> <span class="n">vcf</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-p</span> <span class="pre">vcf</span></code>: input format</li>
</ul>
<p>We can get some quick stats with <code class="docutils literal"><span class="pre">rtg</span> <span class="pre">vcfstats</span></code>:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">rtg</span> <span class="n">vcfstats</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>Example output from <code class="docutils literal"><span class="pre">rtg</span> <span class="pre">vcfstats</span></code>:</p>
<div class="code highlight-default"><div class="highlight"><pre><span></span><span class="n">Location</span>                     <span class="p">:</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
<span class="n">Failed</span> <span class="n">Filters</span>               <span class="p">:</span> <span class="mi">0</span>
<span class="n">Passed</span> <span class="n">Filters</span>               <span class="p">:</span> <span class="mi">516</span>
<span class="n">SNPs</span>                         <span class="p">:</span> <span class="mi">399</span>
<span class="n">MNPs</span>                         <span class="p">:</span> <span class="mi">0</span>
<span class="n">Insertions</span>                   <span class="p">:</span> <span class="mi">104</span>
<span class="n">Deletions</span>                    <span class="p">:</span> <span class="mi">13</span>
<span class="n">Indels</span>                       <span class="p">:</span> <span class="mi">0</span>
<span class="n">Same</span> <span class="k">as</span> <span class="n">reference</span>            <span class="p">:</span> <span class="mi">0</span>
<span class="n">SNP</span> <span class="n">Transitions</span><span class="o">/</span><span class="n">Transversions</span><span class="p">:</span> <span class="mf">1.87</span> <span class="p">(</span><span class="mi">286</span><span class="o">/</span><span class="mi">153</span><span class="p">)</span>
<span class="n">Total</span> <span class="n">Het</span><span class="o">/</span><span class="n">Hom</span> <span class="n">ratio</span>          <span class="p">:</span> <span class="mf">3.20</span> <span class="p">(</span><span class="mi">393</span><span class="o">/</span><span class="mi">123</span><span class="p">)</span>
<span class="n">SNP</span> <span class="n">Het</span><span class="o">/</span><span class="n">Hom</span> <span class="n">ratio</span>            <span class="p">:</span> <span class="mf">8.98</span> <span class="p">(</span><span class="mi">359</span><span class="o">/</span><span class="mi">40</span><span class="p">)</span>
<span class="n">MNP</span> <span class="n">Het</span><span class="o">/</span><span class="n">Hom</span> <span class="n">ratio</span>            <span class="p">:</span> <span class="o">-</span> <span class="p">(</span><span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">)</span>
<span class="n">Insertion</span> <span class="n">Het</span><span class="o">/</span><span class="n">Hom</span> <span class="n">ratio</span>      <span class="p">:</span> <span class="mf">0.30</span> <span class="p">(</span><span class="mi">24</span><span class="o">/</span><span class="mi">80</span><span class="p">)</span>
<span class="n">Deletion</span> <span class="n">Het</span><span class="o">/</span><span class="n">Hom</span> <span class="n">ratio</span>       <span class="p">:</span> <span class="mf">3.33</span> <span class="p">(</span><span class="mi">10</span><span class="o">/</span><span class="mi">3</span><span class="p">)</span>
<span class="n">Indel</span> <span class="n">Het</span><span class="o">/</span><span class="n">Hom</span> <span class="n">ratio</span>          <span class="p">:</span> <span class="o">-</span> <span class="p">(</span><span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">)</span>
<span class="n">Insertion</span><span class="o">/</span><span class="n">Deletion</span> <span class="n">ratio</span>     <span class="p">:</span> <span class="mf">8.00</span> <span class="p">(</span><span class="mi">104</span><span class="o">/</span><span class="mi">13</span><span class="p">)</span>
<span class="n">Indel</span><span class="o">/</span><span class="n">SNP</span><span class="o">+</span><span class="n">MNP</span> <span class="n">ratio</span>          <span class="p">:</span> <span class="mf">0.29</span> <span class="p">(</span><span class="mi">117</span><span class="o">/</span><span class="mi">399</span><span class="p">)</span>
</pre></div>
</div>
<p>However, we can also run <a class="reference external" href="http://www.htslib.org/doc/bcftools.html">BCFtools</a> to extract more detailed statistics about our variant calls:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">bcftools</span> <span class="n">stats</span> <span class="o">-</span><span class="n">F</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="n">final</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span> <span class="o">-</span><span class="n">s</span> <span class="o">-</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">&gt;</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span><span class="o">.</span><span class="n">stats</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-s</span> <span class="pre">-</span></code>: list of samples for sample stats, “-” to include all samples</li>
<li><code class="docutils literal"><span class="pre">-F</span> <span class="pre">FILE</span></code>: faidx indexed reference sequence file to determine INDEL context</li>
</ul>
<p>Now we take the stats and make some plots (e.g. <a class="reference internal" href="#fig-vcfstats"><span class="std std-numref">Fig. 7.2</span></a>) which are particular of interest if having multiple samples, as one can easily compare them. However, we are only working with one here:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">variants</span><span class="o">/</span><span class="n">plots</span>
<span class="n">plot</span><span class="o">-</span><span class="n">vcfstats</span> <span class="o">-</span><span class="n">p</span> <span class="n">variants</span><span class="o">/</span><span class="n">plots</span><span class="o">/</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span><span class="o">.</span><span class="n">stats</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-p</span></code>: The output files prefix, add a slash at the end to create a new directory.</li>
</ul>
<div class="figure" id="id5">
<span id="fig-vcfstats"></span><img alt="../_images/vcfstats.png" src="../_images/vcfstats.png" />
<p class="caption"><span class="caption-number">Fig. 7.2 </span><span class="caption-text">Example of <code class="docutils literal"><span class="pre">plot-vcfstats</span></code> output.</span></p>
</div>
</div>
<div class="section" id="variant-filtration">
<h3>7.8.3. Variant filtration<a class="headerlink" href="#variant-filtration" title="Permalink to this headline">¶</a></h3>
<p>Variant filtration is a big topic in itself <a class="reference internal" href="#olsen2015" id="id2">[OLSEN2015]</a>.
There is no consens yet and research on how to best filter variants is ongoing.</p>
<p>We will do some simple filtration procedures here.
For one, we can filter out low quality reads.</p>
<p>Here, we only include variants that have quality &gt; 30.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># use rtg vcfffilter</span>
<span class="n">rtg</span> <span class="n">vcffilter</span> <span class="o">-</span><span class="n">q</span> <span class="mi">30</span> <span class="o">-</span><span class="n">i</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">o</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">q30</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-i</span> <span class="pre">FILE</span></code>: input file</li>
<li><code class="docutils literal"><span class="pre">-o</span> <span class="pre">FILE</span></code>: output file</li>
<li><code class="docutils literal"><span class="pre">-q</span> <span class="pre">FLOAT</span></code>: minimal allowed quality in output.</li>
</ul>
<p>or use <a class="reference external" href="https://github.com/vcflib/vcflib#vcflib">vcflib</a>:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># or use vcflib</span>
<span class="n">zcat</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>  <span class="o">|</span> <span class="n">vcffilter</span> <span class="o">-</span><span class="n">f</span> <span class="s2">&quot;QUAL &gt;= 30&quot;</span> <span class="o">|</span> <span class="n">gzip</span> <span class="o">&gt;</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">q30</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="n">z</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">-f</span> <span class="pre">&quot;QUAL</span> <span class="pre">&gt;=</span> <span class="pre">30&quot;</span></code>: we only include variants that have been called with quality &gt;= 30.</li>
</ul>
<p>Quick stats for the filtered variants:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># look at stats for filtered</span>
<span class="n">rtg</span> <span class="n">vcfstats</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">mpileup</span><span class="o">.</span><span class="n">q30</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p><a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a> adds some extra information to the vcf-fiels it creates.
This allows for some more detailed filtering.
This strategy will NOT work on the <a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a> mpileup called variants
Here we filter, based on some recommendation form the developer of <a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a>:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">zcat</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">freebayes</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>  <span class="o">|</span> <span class="n">vcffilter</span> <span class="o">-</span><span class="n">f</span> <span class="s2">&quot;QUAL &gt; 1 &amp; QUAL / AO &gt; 10 &amp; SAF &gt; 0 &amp; SAR &gt; 0 &amp; RPR &gt; 1 &amp; RPL &gt; 1&quot;</span> <span class="o">|</span> <span class="n">gzip</span> <span class="o">&gt;</span> <span class="n">variants</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">freebayes</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">QUAL</span> <span class="pre">&gt;</span> <span class="pre">1</span></code>: removes really bad sites</li>
<li><code class="docutils literal"><span class="pre">QUAL</span> <span class="pre">/</span> <span class="pre">AO</span> <span class="pre">&gt;</span> <span class="pre">10</span></code>: additional contribution of each obs should be 10 log units (~ Q10 per read)</li>
<li><code class="docutils literal"><span class="pre">SAF</span> <span class="pre">&gt;</span> <span class="pre">0</span> <span class="pre">&amp;</span> <span class="pre">SAR</span> <span class="pre">&gt;</span> <span class="pre">0</span></code>: reads on both strands</li>
<li><code class="docutils literal"><span class="pre">RPR</span> <span class="pre">&gt;</span> <span class="pre">1</span> <span class="pre">&amp;</span> <span class="pre">RPL</span> <span class="pre">&gt;</span> <span class="pre">1</span></code>: at least two reads “balanced” to each side of the site</li>
</ul>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<p class="last">Look at the statistics. One ratio that is mentioned in the statistics is transition transversion ratio (<em>ts/tv</em>).
Explain what this ratio is and why the observed ratio makes sense.</p>
</div>
<p>This strategy used here will do for our purposes.
However, several more elaborate filtering strategies have been explored, e.g. <a class="reference external" href="https://github.com/ekg/freebayes#observation-filters-and-qualities">here</a>.</p>
<p class="rubric">References</p>
<table class="docutils citation" frame="void" id="nielsen2011" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[NIELSEN2011]</a></td><td>Nielsen R, Paul JS, Albrechtsen A, Song YS. Genotype and SNP calling from next-generation sequencing data. <a class="reference external" href="http://doi.org/10.1038/nrg2986">Nat Rev Genetics, 2011, 12:433-451</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="olsen2015" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[OLSEN2015]</a></td><td>Olsen ND et al. Best practices for evaluating single nucleotide variant calling methods for microbial genomics. <a class="reference external" href="https://doi.org/10.3389/fgene.2015.00235">Front. Genet., 2015, 6:235.</a></td></tr>
</tbody>
</table>
</div>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../ngs-annotation/index.html" class="btn btn-neutral float-right" title="8. Genome annotation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ngs-taxonomic-investigation/index.html" class="btn btn-neutral" title="6. Taxonomic investigation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2018, Sebastian Schmeier (https://sschmeier.com).
      Last updated on Mar 14, 2018.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'2017.10',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>