

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>8. NGS - Variant calling &mdash; Documentation</title>
  

  
  
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../_static/css/seb.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="Documentation" href="../index.html"/>
        <link rel="next" title="9. Quick command reference" href="../general/quickref.html"/>
        <link rel="prev" title="7. NGS - Read mapping" href="../ngs-mapping/index.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> Genomics Tutorial
          

          
            
            <img src="../_static/icon.png" class="logo" />
          
          </a>

          
            
            
              <div class="version">
                2017.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../cli/index.html">1. Command-line interface introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-tools/index.html">2. NGS - Tool installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-qc/index.html">3. NGS - Quality control</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-taxonomic-investigation/index.html">4. NGS - Taxonomic investigation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-assembly/index.html">5. NGS - Genome assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-annotation/index.html">6. NGS - Genome annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-mapping/index.html">7. NGS - Read mapping</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">8. NGS - Variant calling</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preface">8.1. Preface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#before-we-start">8.2. Before we start</a></li>
<li class="toctree-l2"><a class="reference internal" href="#learning-outcomes">8.3. Learning outcomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing-necessary-software">8.4. Installing necessary software</a></li>
<li class="toctree-l2"><a class="reference internal" href="#preprocessing">8.5. Preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#calling-variants">8.6. Calling variants</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#samtools-mpileup">8.6.1. SAMtools mpileup</a></li>
<li class="toctree-l3"><a class="reference internal" href="#freebayes">8.6.2. freebayes</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#post-processing">8.7. Post-processing</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#understanding-the-output-files-vcf">8.7.1. Understanding the output files (.vcf)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#statistics-and-filter">8.7.2. Statistics and filter</a></li>
<li class="toctree-l3"><a class="reference internal" href="#finding-variants-of-interest-vai">8.7.3. Finding variants of interest (VAI)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#overlap-variants-with-genes">8.7.4. Overlap variants with genes</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../general/quickref.html">9. Quick command reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../general/code.html">10. Coding solutions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../general/downloads.html">11. Downloads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../general/references.html">12. Bibliography</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">Genomics Tutorial</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>8. NGS - Variant calling</li>
      <li class="wy-breadcrumbs-aside">
        
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="ngs-variant-calling">
<h1>8. NGS - Variant calling<a class="headerlink" href="#ngs-variant-calling" title="Permalink to this headline">¶</a></h1>
<div class="section" id="preface">
<h2>8.1. Preface<a class="headerlink" href="#preface" title="Permalink to this headline">¶</a></h2>
<p>In this section we will use our genome assembly based on the ancestor and call
genetic variants in the evolved line.</p>
<p>There is an accompanying lecture for this tutorial:</p>
<ul class="simple">
<li><a class="reference external" href="http://dx.doi.org/10.6084/m9.figshare.1515026">SNPs - GWAS - eQTLs introduction</a></li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You will encounter some <strong>To-do</strong> sections at times. Write the solutions and answers into a text-file.</p>
</div>
</div>
<div class="section" id="before-we-start">
<h2>8.2. Before we start<a class="headerlink" href="#before-we-start" title="Permalink to this headline">¶</a></h2>
<p>Lets see how our directory structure looks so far:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>
<span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">F</span>
</pre></div>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">annotation</span><span class="o">/</span>
<span class="n">assembly</span><span class="o">/</span>
<span class="n">data</span><span class="o">/</span>
<span class="n">kraken</span><span class="o">/</span>
<span class="n">mappings</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">++</span>
<span class="n">trimmed</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">fastqc</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">solexaqa</span><span class="o">/</span>
</pre></div>
</div>
</div>
<div class="section" id="learning-outcomes">
<h2>8.3. Learning outcomes<a class="headerlink" href="#learning-outcomes" title="Permalink to this headline">¶</a></h2>
<p>After studying this tutorial you should be able to:</p>
<ol class="arabic simple">
<li>Use tools to call variants based on a reference genome.</li>
<li>Identify variants of interests.</li>
<li>Understand how the variants might affect the observed biology.</li>
</ol>
</div>
<div class="section" id="installing-necessary-software">
<h2>8.4. Installing necessary software<a class="headerlink" href="#installing-necessary-software" title="Permalink to this headline">¶</a></h2>
<p>Tools we are going to use in this section and how to intall them if you not have done it yet.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># activate the env</span>
<span class="n">source</span> <span class="n">activate</span> <span class="n">ngs</span>

<span class="c1"># Install these tools into the conda environment</span>
<span class="c1"># if not already installed</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">samtools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">bamtools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">bcftools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">freebayes</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">rtg</span><span class="o">-</span><span class="n">tools</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">tabix</span>
</pre></div>
</div>
</div>
<div class="section" id="preprocessing">
<h2>8.5. Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">¶</a></h2>
<p>We first need to make an index of our reference genome as this is required by the SNP caller.
Given a scaffold/contig file in fasta-format, e.g. <code class="docutils literal"><span class="pre">scaffolds.fasta</span></code> which is
located ina  directory <code class="docutils literal"><span class="pre">assembly/spades_final</span></code>, use <a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a> to do this:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">faidx</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades_final</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Furthermore we need to pre-process our mapping files a bit further and creaee a bam-index file (<code class="docutils literal"><span class="pre">.bai</span></code>) for each o the bam-files, e.g.:</p>
<p class="sebcode">bamtools index -in mappings/evolved-6.sorted.bam</p>
<p>Lets also create a new directory for the variants:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">variants</span>
</pre></div>
</div>
</div>
<div class="section" id="calling-variants">
<h2>8.6. Calling variants<a class="headerlink" href="#calling-variants" title="Permalink to this headline">¶</a></h2>
<div class="section" id="samtools-mpileup">
<h3>8.6.1. <a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a> mpileup<a class="headerlink" href="#samtools-mpileup" title="Permalink to this headline">¶</a></h3>
<p>We use the sorted bam-file that we produced in the mapping step before.</p>
<p class="sebcode"># We first pile up all the reads
samtools mpileup -g -f assembly/spades_final/scaffolds.fasta mappings/evolved-6.sorted.bam &gt; variants/evolved-6.mpileup.bcf
# Now we call the variants
bcftools view -c -v variants/evolved-6.mpileup.bcf &gt; variants/evolved-6.mpileup.vcf</p>
</div>
<div class="section" id="freebayes">
<h3>8.6.2. <a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a><a class="headerlink" href="#freebayes" title="Permalink to this headline">¶</a></h3>
<p>Now we can do some variant calling with another tool called <a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a>.
Given a reference genome scaffold file in fasta-format, e.g. <code class="docutils literal"><span class="pre">scaffolds.fasta</span></code> and the index in <code class="docutils literal"><span class="pre">.fai</span></code> format and a mapping file (e.g. &quot;evolved-6.sorted.bam&quot;) and a mapping index, we can call <a class="reference external" href="https://github.com/ekg/freebayes">freebayes</a> like so:</p>
<p class="sebcode"># Now we call variants and pipe the results into a new file
freebayes -f assembly/spades_final/scaffolds.fasta mappings/evolved-6.sorted.bam &gt; variants/evolved-6.freebayes.vcf</p>
<p>This will result in a variants file &quot;evolved-6.freebayes.vcf&quot;.</p>
</div>
</div>
<div class="section" id="post-processing">
<h2>8.7. Post-processing<a class="headerlink" href="#post-processing" title="Permalink to this headline">¶</a></h2>
<div class="section" id="understanding-the-output-files-vcf">
<h3>8.7.1. Understanding the output files (.vcf)<a class="headerlink" href="#understanding-the-output-files-vcf" title="Permalink to this headline">¶</a></h3>
<p>Lets look at a vcf-file:</p>
<p class="sebcode"># first 10 lines, which are part of the header
cat variants/evolved-6.freebayes.vcf | head</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1">##fileformat=VCFv4.1</span>
<span class="c1">##fileDate=20161122</span>
<span class="c1">##source=freeBayes v1.0.2-29-g41c1313</span>
<span class="c1">##reference=genome/scaffolds.fasta</span>
<span class="c1">##contig=&lt;ID=NODE_1_length_1394677_cov_15.3771,length=1394677&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_2_length_1051867_cov_15.4779,length=1051867&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_3_length_950567_cov_15.4139,length=950567&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_4_length_925223_cov_15.3905,length=925223&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_5_length_916389_cov_15.4457,length=916389&gt;</span>
<span class="c1">##contig=&lt;ID=NODE_6_length_772252_cov_15.4454,length=772252&gt;</span>
</pre></div>
</div>
<p>Lets look at the variants:</p>
<p class="sebcode"># remove header lines and look at top 4 entires
cat variants/evolved-6.freebayes.vcf | egrep -v '##' | head -4</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1">#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  unknown</span>
<span class="n">NODE_1_length_1394677_cov_15</span><span class="o">.</span><span class="mi">3771</span>       <span class="mi">137621</span>  <span class="o">.</span>       <span class="n">T</span>       <span class="n">C</span>       <span class="mf">76.5197</span> <span class="o">.</span>       <span class="n">AB</span><span class="o">=</span><span class="mf">0.318182</span><span class="p">;</span><span class="n">ABP</span><span class="o">=</span><span class="mf">9.32731</span><span class="p">;</span><span class="n">AC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mf">0.5</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">AO</span><span class="o">=</span><span class="mi">7</span><span class="p">;</span><span class="n">CIGAR</span><span class="o">=</span><span class="mi">1</span><span class="n">X</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">22</span><span class="p">;</span><span class="n">DPB</span><span class="o">=</span><span class="mi">22</span><span class="p">;</span><span class="n">DPRA</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">EPP</span><span class="o">=</span><span class="mf">18.2106</span><span class="p">;</span><span class="n">EPPR</span><span class="o">=</span><span class="mf">4.31318</span><span class="p">;</span><span class="n">GTI</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">LEN</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MEANALT</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MQM</span><span class="o">=</span><span class="mf">56.1429</span><span class="p">;</span><span class="n">MQMR</span><span class="o">=</span><span class="mf">56.4</span><span class="p">;</span><span class="n">NS</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">NUMALT</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">ODDS</span><span class="o">=</span><span class="mf">17.6193</span><span class="p">;</span><span class="n">PAIRED</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">PAIREDR</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">PAO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PQA</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PQR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PRO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">QA</span><span class="o">=</span><span class="mi">268</span><span class="p">;</span><span class="n">QR</span><span class="o">=</span><span class="mi">540</span><span class="p">;</span><span class="n">RO</span><span class="o">=</span><span class="mi">15</span><span class="p">;</span><span class="n">RPL</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">RPP</span><span class="o">=</span><span class="mf">18.2106</span><span class="p">;</span><span class="n">RPPR</span><span class="o">=</span><span class="mf">6.62942</span><span class="p">;</span><span class="n">RPR</span><span class="o">=</span><span class="mi">7</span><span class="p">;</span><span class="n">RUN</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">SAF</span><span class="o">=</span><span class="mi">7</span><span class="p">;</span><span class="n">SAP</span><span class="o">=</span><span class="mf">18.2106</span><span class="p">;</span><span class="n">SAR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">SRF</span><span class="o">=</span><span class="mi">12</span><span class="p">;</span><span class="n">SRP</span><span class="o">=</span><span class="mf">14.7363</span><span class="p">;</span><span class="n">SRR</span><span class="o">=</span><span class="mi">3</span><span class="p">;</span><span class="n">TYPE</span><span class="o">=</span><span class="n">snp</span>       <span class="n">GT</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">DPR</span><span class="p">:</span><span class="n">RO</span><span class="p">:</span><span class="n">QR</span><span class="p">:</span><span class="n">AO</span><span class="p">:</span><span class="n">QA</span><span class="p">:</span><span class="n">GL</span>    <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">22</span><span class="p">:</span><span class="mi">22</span><span class="p">,</span><span class="mi">7</span><span class="p">:</span><span class="mi">15</span><span class="p">:</span><span class="mi">540</span><span class="p">:</span><span class="mi">7</span><span class="p">:</span><span class="mi">268</span><span class="p">:</span><span class="o">-</span><span class="mf">17.3644</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="o">-</span><span class="mf">42.2185</span>
<span class="n">NODE_1_length_1394677_cov_15</span><span class="o">.</span><span class="mi">3771</span>       <span class="mi">568696</span>  <span class="o">.</span>       <span class="n">G</span>       <span class="n">A</span>       <span class="mf">1269.62</span> <span class="o">.</span>       <span class="n">AB</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ABP</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">AC</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">AO</span><span class="o">=</span><span class="mi">38</span><span class="p">;</span><span class="n">CIGAR</span><span class="o">=</span><span class="mi">1</span><span class="n">X</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">38</span><span class="p">;</span><span class="n">DPB</span><span class="o">=</span><span class="mi">38</span><span class="p">;</span><span class="n">DPRA</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">EPP</span><span class="o">=</span><span class="mf">3.23888</span><span class="p">;</span><span class="n">EPPR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">GTI</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">LEN</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MEANALT</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MQM</span><span class="o">=</span><span class="mi">60</span><span class="p">;</span><span class="n">MQMR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">NS</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">NUMALT</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">ODDS</span><span class="o">=</span><span class="mf">57.2844</span><span class="p">;</span><span class="n">PAIRED</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">PAIREDR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PAO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PQA</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PQR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PRO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">QA</span><span class="o">=</span><span class="mi">1438</span><span class="p">;</span><span class="n">QR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">RO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">RPL</span><span class="o">=</span><span class="mi">20</span><span class="p">;</span><span class="n">RPP</span><span class="o">=</span><span class="mf">3.23888</span><span class="p">;</span><span class="n">RPPR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">RPR</span><span class="o">=</span><span class="mi">18</span><span class="p">;</span><span class="n">RUN</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">SAF</span><span class="o">=</span><span class="mi">20</span><span class="p">;</span><span class="n">SAP</span><span class="o">=</span><span class="mf">3.23888</span><span class="p">;</span><span class="n">SAR</span><span class="o">=</span><span class="mi">18</span><span class="p">;</span><span class="n">SRF</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">SRP</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">SRR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">TYPE</span><span class="o">=</span><span class="n">snp</span>      <span class="n">GT</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">DPR</span><span class="p">:</span><span class="n">RO</span><span class="p">:</span><span class="n">QR</span><span class="p">:</span><span class="n">AO</span><span class="p">:</span><span class="n">QA</span><span class="p">:</span><span class="n">GL</span>        <span class="mi">1</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">38</span><span class="p">:</span><span class="mi">38</span><span class="p">,</span><span class="mi">38</span><span class="p">:</span><span class="mi">0</span><span class="p">:</span><span class="mi">0</span><span class="p">:</span><span class="mi">38</span><span class="p">:</span><span class="mi">1438</span><span class="p">:</span><span class="o">-</span><span class="mf">129.701</span><span class="p">,</span><span class="o">-</span><span class="mf">11.4391</span><span class="p">,</span><span class="mi">0</span>
<span class="n">NODE_1_length_1394677_cov_15</span><span class="o">.</span><span class="mi">3771</span>       <span class="mi">612771</span>  <span class="o">.</span>       <span class="n">T</span>       <span class="n">C</span>       <span class="mf">60.7485</span> <span class="o">.</span>       <span class="n">AB</span><span class="o">=</span><span class="mf">0.3</span><span class="p">;</span><span class="n">ABP</span><span class="o">=</span><span class="mf">9.95901</span><span class="p">;</span><span class="n">AC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mf">0.5</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">AO</span><span class="o">=</span><span class="mi">6</span><span class="p">;</span><span class="n">CIGAR</span><span class="o">=</span><span class="mi">1</span><span class="n">X</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">20</span><span class="p">;</span><span class="n">DPB</span><span class="o">=</span><span class="mi">20</span><span class="p">;</span><span class="n">DPRA</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">EPP</span><span class="o">=</span><span class="mf">4.45795</span><span class="p">;</span><span class="n">EPPR</span><span class="o">=</span><span class="mf">8.59409</span><span class="p">;</span><span class="n">GTI</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">LEN</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MEANALT</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MQM</span><span class="o">=</span><span class="mf">49.5</span><span class="p">;</span><span class="n">MQMR</span><span class="o">=</span><span class="mf">54.3571</span><span class="p">;</span><span class="n">NS</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">NUMALT</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">ODDS</span><span class="o">=</span><span class="mf">13.9879</span><span class="p">;</span><span class="n">PAIRED</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">PAIREDR</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">PAO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PQA</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PQR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">PRO</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">QA</span><span class="o">=</span><span class="mi">223</span><span class="p">;</span><span class="n">QR</span><span class="o">=</span><span class="mi">540</span><span class="p">;</span><span class="n">RO</span><span class="o">=</span><span class="mi">14</span><span class="p">;</span><span class="n">RPL</span><span class="o">=</span><span class="mi">6</span><span class="p">;</span><span class="n">RPP</span><span class="o">=</span><span class="mf">16.0391</span><span class="p">;</span><span class="n">RPPR</span><span class="o">=</span><span class="mf">33.4109</span><span class="p">;</span><span class="n">RPR</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">RUN</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">SAF</span><span class="o">=</span><span class="mi">4</span><span class="p">;</span><span class="n">SAP</span><span class="o">=</span><span class="mf">4.45795</span><span class="p">;</span><span class="n">SAR</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">SRF</span><span class="o">=</span><span class="mi">4</span><span class="p">;</span><span class="n">SRP</span><span class="o">=</span><span class="mf">8.59409</span><span class="p">;</span><span class="n">SRR</span><span class="o">=</span><span class="mi">10</span><span class="p">;</span><span class="n">TYPE</span><span class="o">=</span><span class="n">snp</span>    <span class="n">GT</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">DPR</span><span class="p">:</span><span class="n">RO</span><span class="p">:</span><span class="n">QR</span><span class="p">:</span><span class="n">AO</span><span class="p">:</span><span class="n">QA</span><span class="p">:</span><span class="n">GL</span>        <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">20</span><span class="p">:</span><span class="mi">20</span><span class="p">,</span><span class="mi">6</span><span class="p">:</span><span class="mi">14</span><span class="p">:</span><span class="mi">540</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">223</span><span class="p">:</span><span class="o">-</span><span class="mf">12.5734</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="o">-</span><span class="mf">40.0605</span>
</pre></div>
</div>
</div>
<div class="section" id="statistics-and-filter">
<h3>8.7.2. Statistics and filter<a class="headerlink" href="#statistics-and-filter" title="Permalink to this headline">¶</a></h3>
<p>Now we can use it to do some statistics and filter our variant calls.</p>
<p class="sebcode"># get statistics
rtg vcfstats variants/evolved-6.freebayes.vcf</p>
<p class="sebcode"># only keep entries with qual of min 30
rtg vcffilter -q 30 -i variants/evolved-6.freebayes.vcf -o variants/evolved-6.freebayes-q30.vcf</p>
<p class="sebcode"># look at stats for filtered
rtg vcfstats variants/evolved-6.freebayes-q30.vcf</p>
</div>
<div class="section" id="finding-variants-of-interest-vai">
<h3>8.7.3. Finding variants of interest (VAI)<a class="headerlink" href="#finding-variants-of-interest-vai" title="Permalink to this headline">¶</a></h3>
<p>Things to consider when looking for VAI:</p>
<ul class="simple">
<li>The quality score of the variant call.<ul>
<li>Do we call the variant with a higher then normal score?</li>
</ul>
</li>
<li>The mapping quality score.<ul>
<li>How confident are we that the reads where mapped here correctly?</li>
</ul>
</li>
<li>The location of the SNP.<ul>
<li>SNPs in larger contigs probably more interesting than in tiny contigs.</li>
<li>Does the SNP overlap a coding region in the genome annotation?</li>
</ul>
</li>
<li>The type of SNP.</li>
</ul>
</div>
<div class="section" id="overlap-variants-with-genes">
<h3>8.7.4. Overlap variants with genes<a class="headerlink" href="#overlap-variants-with-genes" title="Permalink to this headline">¶</a></h3>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../general/quickref.html" class="btn btn-neutral float-right" title="9. Quick command reference" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ngs-mapping/index.html" class="btn btn-neutral" title="7. NGS - Read mapping" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2017, Sebastian Schmeier.
      Last updated on Jan 30, 2017.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'2017.1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>