

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>9. Orthology and Phylogeny</title>
  

  
  
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/seb.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="10. Variants-of-interest" href="../ngs-voi/index.html" />
    <link rel="prev" title="8. Genome annotation" href="../ngs-annotation/index.html" />
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121349593-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121349593-1');
</script>


  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> Genomics Tutorial
          

          
            
            <img src="../_static/icon.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                2019.03
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">1. Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-tools/index.html">2. Tool installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-qc/index.html">3. Quality control</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-assembly/index.html">4. Genome assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-mapping/index.html">5. Read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-taxonomic-investigation/index.html">6. Taxonomic investigation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-variantcalling/index.html">7. Variant calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-annotation/index.html">8. Genome annotation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">9. Orthology and Phylogeny</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preface">9.1. Preface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#learning-outcomes">9.2. Learning outcomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#before-we-start">9.3. Before we start</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing-the-software">9.4. Installing the software</a></li>
<li class="toctree-l2"><a class="reference internal" href="#finding-orthologues-using-blast">9.5. Finding orthologues using BLAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="#performing-an-alignment">9.6. Performing an alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#building-a-phylogeny">9.7. Building a phylogeny</a></li>
<li class="toctree-l2"><a class="reference internal" href="#visualizing-the-phylogeny">9.8. Visualizing the phylogeny</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-voi/index.html">10. Variants-of-interest</a></li>
<li class="toctree-l1"><a class="reference internal" href="../quickref.html">11. Quick command reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../code.html">12. Coding solutions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../downloads.html">13. Downloads</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Genomics Tutorial</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>9. Orthology and Phylogeny</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="orthology-and-phylogeny">
<span id="ngs-orthology"></span><h1>9. Orthology and Phylogeny<a class="headerlink" href="#orthology-and-phylogeny" title="Permalink to this headline">¶</a></h1>
<div class="section" id="preface">
<h2>9.1. Preface<a class="headerlink" href="#preface" title="Permalink to this headline">¶</a></h2>
<p>In this section you will use some software to find orthologue genes and do phylogenetic reconstructions.</p>
</div>
<div class="section" id="learning-outcomes">
<h2>9.2. Learning outcomes<a class="headerlink" href="#learning-outcomes" title="Permalink to this headline">¶</a></h2>
<p>After studying this tutorial you should be able to:</p>
<ol class="arabic simple">
<li>Use bioinformatics software to find orthologues in the NCBI database.</li>
<li>Use bioinformatics software to perform sequence alignment.</li>
<li>Use bioinformatics software to perform phylogenetic reconstructions.</li>
</ol>
</div>
<div class="section" id="before-we-start">
<h2>9.3. Before we start<a class="headerlink" href="#before-we-start" title="Permalink to this headline">¶</a></h2>
<p>Lets see how our directory structure looks so far:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>
<span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">F</span>
</pre></div>
</div>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">annotation</span><span class="o">/</span>
<span class="n">assembly</span><span class="o">/</span>
<span class="n">data</span><span class="o">/</span>
<span class="n">kraken</span><span class="o">/</span>
<span class="n">mappings</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">fastqc</span><span class="o">/</span>
<span class="n">variants</span><span class="o">/</span>
</pre></div>
</div>
<p>Make a directory for the phylogeny results (in your analysis directory):</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">phylogeny</span>
</pre></div>
</div>
<p>Download the fasta file of the <em>S. cerevisiase</em> TEF2 gene to the phylogeny folder:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="n">phylogeny</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">O</span> <span class="n">http</span><span class="p">:</span><span class="o">//</span><span class="n">compbio</span><span class="o">.</span><span class="n">massey</span><span class="o">.</span><span class="n">ac</span><span class="o">.</span><span class="n">nz</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="mi">203341</span><span class="o">/</span><span class="n">s_cerev_tef2</span><span class="o">.</span><span class="n">fas</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Should the download fail, download manually from <a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a>.</p>
</div>
</div>
<div class="section" id="installing-the-software">
<h2>9.4. Installing the software<a class="headerlink" href="#installing-the-software" title="Permalink to this headline">¶</a></h2>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># activate the env</span>
<span class="n">conda</span> <span class="n">activate</span> <span class="n">ngs</span>

<span class="n">conda</span> <span class="n">install</span> <span class="n">blast</span>
</pre></div>
</div>
<p>This will install a <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> executable that you can use to remotely query the NCBI database.</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="n">muscle</span>
</pre></div>
</div>
<p>This will install <a class="reference external" href="http://www.ebi.ac.uk/Tools/msa/muscle/">MUSCLE</a>, alignment program that you can use to align nucleotide or protein sequences.</p>
<p>We will also install <a class="reference external" href="https://github.com/amkozlov/raxml-ng">RAxML-NG</a>, a phylogenetic tree inference tool, which uses
maximum-likelihood (ML) optimality criterion. However, there is no conda
repository for it yet. Thus, we need to download it manually.</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wget</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">amkozlov</span><span class="o">/</span><span class="n">raxml</span><span class="o">-</span><span class="n">ng</span><span class="o">/</span><span class="n">releases</span><span class="o">/</span><span class="n">download</span><span class="o">/</span><span class="mf">0.5</span><span class="o">.</span><span class="mi">1</span><span class="o">/</span><span class="n">raxml</span><span class="o">-</span><span class="n">ng_v0</span><span class="o">.</span><span class="mf">5.1</span><span class="n">b_linux_x86_64</span><span class="o">.</span><span class="n">zip</span>

<span class="n">unzip</span> <span class="n">raxml</span><span class="o">-</span><span class="n">ng_v0</span><span class="o">.</span><span class="mf">5.1</span><span class="n">b_linux_x86_64</span><span class="o">.</span><span class="n">zip</span>

<span class="n">rm</span> <span class="n">raxml</span><span class="o">-</span><span class="n">ng_v0</span><span class="o">.</span><span class="mf">5.1</span><span class="n">b_linux_x86_64</span><span class="o">.</span><span class="n">zip</span>
</pre></div>
</div>
</div>
<div class="section" id="finding-orthologues-using-blast">
<h2>9.5. Finding orthologues using BLAST<a class="headerlink" href="#finding-orthologues-using-blast" title="Permalink to this headline">¶</a></h2>
<p>We will first make a <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> database of our current assembly so that we can
find the orthologous sequence of the <em>S. cerevisiae</em> gene.
To do this, we run the command <code class="docutils literal notranslate"><span class="pre">makeblastdb</span></code>:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span># create blast db
makeblastdb –in ../assembly/spades_final/scaffolds.fasta –dbtype nucl
</pre></div>
</div>
<p>To run <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a>, we give it:</p>
<ul class="simple">
<li><code class="docutils literal notranslate"><span class="pre">-db</span></code>: The name of the database that we are BLASTing</li>
<li><code class="docutils literal notranslate"><span class="pre">-query</span></code>: A fasta format input file</li>
<li>A name for the output files</li>
<li>Some notes about the format we want</li>
</ul>
<p>First, we blast without any formatting:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span>blastn –db ../assembly/spades_final/scaffolds.fasta –query s_cerev_tef2.fas &gt; blast.out
</pre></div>
</div>
<p>This should output a file with a set of <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> hits similar to what you might
see on the <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> web site.</p>
<p>Read through the output (e.g. using <code class="docutils literal notranslate"><span class="pre">nano</span></code>) to see what the results of your <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> run was.</p>
<p>Next we will format the output a little so that it is easier to deal with.</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span>blastn –db ../assembly/spades_final/scaffolds.fasta –query s_cerev_tef2.fas –evalue 1e-100 –outfmt “6 length sseq” &gt; blast_formatted.out
</pre></div>
</div>
<p>This will yield a file that has only the sequences of the subject, so that we can later add those to other fasta files.
However, the formatting is not perfect.
To adjust the format such that it is fasta format, open the file in an editor (e.g. <code class="docutils literal notranslate"><span class="pre">nano</span></code>) and edit the first line so that it has a name for your sequence.
You should know the general format of a fasta-file (e.g. the first line start with a “&gt;”).</p>
<div class="admonition hint">
<p class="first admonition-title">Hint</p>
<p class="last">To edit in <code class="docutils literal notranslate"><span class="pre">vi</span></code> editor, you will need to press the escape key and “a” or “e”.
To save in <code class="docutils literal notranslate"><span class="pre">vi</span></code>, you will need to press the escape key and “w” (write).
To quit <code class="docutils literal notranslate"><span class="pre">vi</span></code>, you will need to press the escape key and “q” (quit).</p>
</div>
<p>Next, you have to replace the dashes (signifying indels in the <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> result).
This can easily be done in <code class="docutils literal notranslate"><span class="pre">vi</span></code>:
Press the escape key, followed by: <code class="docutils literal notranslate"><span class="pre">:%s/\-//g</span></code></p>
<p>Now we will <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> a remote database to get a list of hits that are already in the NCBI database.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">It turns out you may not be able to access this database from within BioLinux. In such a case, download the file named <code class="docutils literal notranslate"><span class="pre">blast.fas</span></code> and place it into your <code class="docutils literal notranslate"><span class="pre">~/analysis/phylogeny/</span></code> directory.</p>
</div>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">O</span> <span class="n">http</span><span class="p">:</span><span class="o">//</span><span class="n">compbio</span><span class="o">.</span><span class="n">massey</span><span class="o">.</span><span class="n">ac</span><span class="o">.</span><span class="n">nz</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="mi">203341</span><span class="o">/</span><span class="n">blast_u</span><span class="o">.</span><span class="n">fas</span>
</pre></div>
</div>
<p>Append the fasta file of your yeast sequence to this file, using whatever set of commands you wish/know.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Should the download fail, download manually from <a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a>.</p>
</div>
</div>
<div class="section" id="performing-an-alignment">
<h2>9.6. Performing an alignment<a class="headerlink" href="#performing-an-alignment" title="Permalink to this headline">¶</a></h2>
<p>We will use <a class="reference external" href="http://www.ebi.ac.uk/Tools/msa/muscle/">MUSCLE</a> to perform our alignment on all the sequences in the <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> fasta file.
This syntax is very simple (change the filenames accordingly):</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span>muscle –in infile.fas –out your_alignment.aln
</pre></div>
</div>
</div>
<div class="section" id="building-a-phylogeny">
<h2>9.7. Building a phylogeny<a class="headerlink" href="#building-a-phylogeny" title="Permalink to this headline">¶</a></h2>
<p>We will use <a class="reference external" href="https://github.com/amkozlov/raxml-ng">RAxML-NG</a> to build our phylogeny.
This uses a maximum likelihood method to infer parameters of evolution and the topology of the tree.
Again, the syntx of the command is fairly simple, except you must make sure that you are using the directory in which <a class="reference external" href="https://github.com/amkozlov/raxml-ng">RAxML-NG</a> sits.</p>
<p>The arguments are:</p>
<ul class="simple">
<li><code class="docutils literal notranslate"><span class="pre">-s</span></code>: an alignment file</li>
<li><code class="docutils literal notranslate"><span class="pre">-m</span></code>: a model of evolution. In this case we will use a general time reversible model with gamma distributed rates (GTR+GAMMA)</li>
<li><code class="docutils literal notranslate"><span class="pre">-n</span></code>: outfile-name</li>
<li><code class="docutils literal notranslate"><span class="pre">-p</span></code>: specify a random number seed for the parsimony inferences</li>
</ul>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span>raxmlHPC -s your_alignment.aln -m GTRGAMMA –n yeast_tree –p 12345
</pre></div>
</div>
</div>
<div class="section" id="visualizing-the-phylogeny">
<h2>9.8. Visualizing the phylogeny<a class="headerlink" href="#visualizing-the-phylogeny" title="Permalink to this headline">¶</a></h2>
<p>We will use the online software <a class="reference external" href="http://itol.embl.de/upload.cgi">Interactive Tree of Life (iTOL)</a> to visualize the tree.
Navigate to this homepage.
Open the file containing your tree (<code class="docutils literal notranslate"><span class="pre">*bestTree.out</span></code>), copy the contents, and paste into the web page (in the Tree text box).</p>
<p>You should then be able to zoom in and out to see where your yeast taxa is.
To find out the closest relative, you will have to use the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/Taxonomy/TaxIdentifier/tax_identifier.cgi">NCBI taxa page</a>.</p>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<p class="last">Are you certain that the yeast are related in the way that the phylogeny suggests? Why might the topology of this phylogeny not truly reflect the evolutionary history of these yeast species?</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../ngs-voi/index.html" class="btn btn-neutral float-right" title="10. Variants-of-interest" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ngs-annotation/index.html" class="btn btn-neutral" title="8. Genome annotation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2019, Sebastian Schmeier (https://sschmeier.com).
      Last updated on Apr 09, 2019.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'2019.03',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="../_static/language_data.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  <script type="text/javascript" src="../_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>